CMakeLists.txt

```diff
@@ -13,21 +13,40 @@ if(NOT COMMAND add_llvm_library)
         # We need to match the build environment for LLVM:
         # In particular, we need C++11 and the -fno-rtti flag
         set(CMAKE_CXX_STANDARD 14)
-	if(CMAKE_BUILD_TYPE MATCHES "Debug")
-		set(CMAKE_CXX_FLAGS "-fPIC -std=gnu++14 -g -O0 -fno-rtti -Wno-deprecated -DHAVE_IEEE_754 -DSIZEOF_VOID_P=8 -DSIZEOF_LONG=8")
-	else()
-		set(CMAKE_CXX_FLAGS "-fPIC -std=gnu++14 -O3 -fno-rtti -Wno-deprecated -DHAVE_IEEE_754 -DSIZEOF_VOID_P=8 -DSIZEOF_LONG=8")
-	endif()
-	set(CMAKE_C_FLAGS "-fPIC")
-
-	find_package(LLVM REQUIRED CONFIG)
+        set(CMAKE_CXX_STANDARD_REQUIRED ON)
+        # add -std=gnu++14
+        set(CMAKE_CXX_EXTENSIONS ON)
+
+        add_compile_definitions(HAVE_IEEE_754)
+        add_compile_definitions(SIZEOF_VOID_P=8)
+        add_compile_definitions(SIZEOF_LONG=8)
+
+        set(COMMON_FLAGS "-Wno-deprecated -Werror")
+        set(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} ${COMMON_FLAGS} -fno-rtti")
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${COMMON_FLAGS}")
+
+        if(CMAKE_BUILD_TYPE MATCHES "Debug")
+		  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O0")
+          set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O0")
+	    else()
+		  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O3")
+          set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O3")
+	    endif()
```

Comment:
would be nice to adjust these indents.

Reply:
It is ready to merge once you fix these indents. It looks not properly aligned. Thanks

Reply:
Updated.